containers/utils/detect_image.py

```diff
@@ -0,0 +1,147 @@
+        ])
+
+
+def draw_objects(draw, objs, labels):
+    """Draws the bounding box and label for each object."""
+    for obj in objs:
+        bbox = obj.bbox
+        draw.rectangle([(bbox.xmin, bbox.ymin), (bbox.xmax, bbox.ymax)],
+                       outline='red')
+        draw.text((bbox.xmin + 10, bbox.ymin + 10),
+                  '%s\n%.2f' % (labels.get(obj.id, obj.id), obj.score),
+                  fill='red')
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-m', '--model', required=True,
+                        help='File path of .tflite file.')
+    parser.add_argument('-i', '--input', required=True,
+                        help='File path of image to process.')
+    parser.add_argument('-l', '--labels',
+                        help='File path of labels file.')
+    parser.add_argument('-t', '--threshold', type=float, default=0.4,
+                        help='Score threshold for detected objects.')
+    parser.add_argument('-o', '--output',
+                        help='File path for the result image with annotations')
+    parser.add_argument('-n', '--team', type=int,
+                        help='Team number')
+    args = parser.parse_args()
+
+    # Init cameras
+    WIDTH, HEIGHT = 720, 1920
+    print("Connecting to Network Tables")
+    ntinst = NetworkTablesInstance.getDefault()
+    ntinst.startClientTeam(args.team)
+    # """Format of these entries found in WPILib documentation."""
+    # detections_entry = ntinst.getTable("ML").getEntry("detections")
+    # print("Starting camera server")
+    cs = CameraServer.getInstance()
+    camera = cs.startAutomaticCapture()
+    camera.setResolution(WIDTH, HEIGHT)
+    cv_sink = cs.getVideo()
+    img = np.zeros(shape=(HEIGHT, WIDTH, 3), dtype=np.uint8)
+    # output = cs.putVideo("MLOut", WIDTH, HEIGHT)
+
+    labels = load_labels(args.labels) if args.labels else {}
+    interpreter = make_interpreter(args.model)
+    interpreter.allocate_tensors()
+
+    if args.output:
+        fourcc = cv2.VideoWriter_fourcc(*'avc1')
+        video = cv2.VideoWriter(args.output + ".mp4", fourcc, 10, (WIDTH, HEIGHT))
+
+    while True:
+        if keyboard.is_pressed('q'):
+            print("Releasing video")
+            video.release()
+            break
+
+        t, frame = cv_sink.grabFrame(img)
+        image = Image.fromarray(frame)
+        scale = detect.set_input(interpreter, image.size,
+                                 lambda size: image.resize(size, Image.ANTIALIAS))
+        interpreter.invoke()
+        objs = detect.get_output(interpreter, args.threshold, scale)
+
+        print('-------RESULTS--------')
+        if not objs:
+            print('No objects detected')
+
+        for obj in objs:
+            print(labels.get(obj.id, obj.id))
+            print('  id:    ', obj.id)
+            print('  score: ', obj.score)
+            print('  bbox:  ', obj.bbox)
+
```

Comment:
needs networktables, obviously. Should be sending a JSON string of all data for current image.